.github/workflows/upgrade-python-requirements.yml

```diff
@@ -50,7 +50,7 @@ jobs:
       - name: setup target branch
         env:
           BRANCH: ${{ inputs.branch }}
-          DEFAULT_BRANCH: ${{ github.event.repository.default_branch }}
+          DEFAULT_BRANCH: ${{ github.event.inputs.default_branch }}
```

Comment:
Upgrade Requirement Jobs have been failing for quite a long time due to this bug in our workflow. Params provided to the workflow can be accessed from `github.event.inputs` and not `github.events.repository`.
For Reference: https://github.blog/changelog/2020-07-06-github-actions-manual-triggers-with-workflow_dispatch/

Reply:
@aht007, can you help me understand the problem?  The only custom input that should be considered here is `inputs.branch`.  The `DEFAULT_BRANCH` env variable is only a fallback in case the calling workflow didn't specify `inputs.branch`, and instead takes its value from `github.event.repository.default_branch`, which [is available on the `schedule` event](https://github.com/dhimmel/dump-actions-context/actions/runs/3283984929/jobs/5409384642#step:2:41).

`DEFAULT_BRANCH` is actually only a last resort.  If you look at the [corresponding workflow template](https://github.com/openedx/.github/blob/master/workflow-templates/upgrade-python-requirements.yml#L20), this workflow need only be called with `branch`.

Can you point me to an example of a calling workflow that is failing?  I suspect that is where the fix should happen.

Reply:
Oh I thought that the repository thing was there by some mistake. Thanks for clarification.
Actually the even the DEFAULT_BRANCH thing is not working correctly under current setup since the `setup target branch` step gives us nothing and hence the workflow fails. For reference you can look below at the failed run.
https://github.com/edx/edx-e2e-tests/actions/runs/3544876433/jobs/5952541056#step:2:2

Reply:
@aht007, this bug was introduced [here](https://github.com/openedx/.github/pull/28).  One can't have naked variables in `[ ]` evaluation because you run into syntax errors if the variable is ever empty.  See my take on  what would fix it:

https://github.com/openedx/.github/pull/35